addon/utils/should-add-token.js

```diff
@@ -1,36 +1,30 @@
-export default function shouldAddToken (url, portalId = '', portalHost) {
-  // if this is a hosted service that belongs to the user's organization
-  // if this is a service with stored credentials
-  const id = portalId.toLowerCase();
-  if (portalHost) {
-    return hostsMatch(portalHost, url);
-  } else {
-    return (
-      !!url.toLowerCase().match('arcgis.com/arcgis/rest/services') ||
-      !!url.toLowerCase().match(`arcgis.com/${id}/arcgis/rest`) ||
-      /utility\w+\.arcgis\.com\/sharing\/servers\/\w{32}\/rest\/services\/Hosted/i.test(url)
-    );
+export default function shouldAddToken (url, serverInfo, portalInfo) {
+  let hostsDidMatch = false;
+  if (portalInfo.portalHostname) {
+    // make sure either the server and the portal are on the same domain
+    hostsDidMatch = hostsMatch(portalInfo.portalHostname, serverInfo.owningSystemUrl);
```

Comment:
do we need any special logic for e.g. www.arcgis.com and servicesx.arcgis.com, utilitiesx.arcgis.com, landscape.arcgis.com etc. or are those included in the authorized cross origin list?

Reply:
I think owningSystemUrl takes care of that. I added some test cases to illustrate.